packages/core/src/external/commonwell/document/document-file-type-detector.ts

```diff
@@ -0,0 +1,88 @@
+ * Magic numbers are unique sequences of bytes that identify the file format or protocol.
+ * In this case, the function reads the first 5 bytes (the magic number) of the file buffer and
+ * matches it against known file type headers.
+ *
+ * @param {Buffer} fileBuffer - The `fileBuffer` parameter is a `Buffer` object that represents the
+ * contents of a file. The first 5 bytes of this buffer, which generally contain the magic number, are used to
+ * identify the file type.
+ * @returns The function `detectFileType` returns a string representing the detected file type.
+ */
+export function detectFileType(fileBuffer: Buffer): string {
+  console.log(`Detecting file type for file: ${fileBuffer.slice(0, 10).toString("hex")}`);
+  if (
+    (fileBuffer[0] === 0x49 &&
+      fileBuffer[1] === 0x49 &&
+      fileBuffer[2] === 0x2a &&
+      fileBuffer[3] === 0x00) ||
+    (fileBuffer[0] === 0x4d &&
+      fileBuffer[1] === 0x4d &&
+      fileBuffer[2] === 0x00 &&
+      fileBuffer[3] === 0x2a)
+  ) {
+    console.log(`Detected file type: image/tiff`);
+    return "image/tiff";
+  } else if (
+    fileBuffer[0] === 0x25 &&
+    fileBuffer[1] === 0x50 &&
+    fileBuffer[2] === 0x44 &&
+    fileBuffer[3] === 0x46 &&
+    fileBuffer[4] === 0x2d
+  ) {
+    console.log(`Detected file type: application/pdf`);
+    return "application/pdf";
+  } else if (
+    fileBuffer[0] === 0x3c &&
+    fileBuffer[1] === 0x3f &&
+    fileBuffer[2] === 0x78 &&
+    fileBuffer[3] === 0x6d &&
+    fileBuffer[4] === 0x6c &&
+    fileBuffer[5] === 0x20
+  ) {
+    console.log(`Detected file type: text/xml`);
+    return "application/xml";
+  } else if (
+    fileBuffer[0] === 0x89 &&
+    fileBuffer[1] === 0x50 &&
+    fileBuffer[2] === 0x4e &&
+    fileBuffer[3] === 0x47
+  ) {
+    console.log(`Detected file type: image/png`);
+    return "image/png";
+  } else if (fileBuffer[0] === 0xff && fileBuffer[1] === 0xd8 && fileBuffer[2] === 0xff) {
+    console.log(`Detected file type: image/jpeg`);
+    return "image/jpeg";
+  } else if (fileBuffer[0] === 0x42 && fileBuffer[1] === 0x4d) {
+    console.log(`Detected file type: image/bmp`);
+    return "image/bmp";
+  } else {
+    throw new Error(`Unknown file type. Cannot convert document: ${fileBuffer.slice(0, 10)}`);
+  }
+}
+/**
+ * The function checks if the MIME type of a document is accepted based on a predefined list of
+ * accepted content types. Returns true if accepted
+ * @param {Document} document - A document object that contains the MIME type of the document.
+ * @returns The function isContentTypeAccepted is returning a boolean value.
+ */
+
+export function isContentTypeAccepted(document: Document): boolean {
+  const acceptedContentTypes = [
+    "image/tiff",
+    "image/tif",
+    "text/xml",
+    "application/xml",
+    "application/pdf",
+    "image/png",
+    "image/jpeg",
+    "image/jpg",
+    "image/bmp",
+    "application/json",
+    "text/html",
+  ];
```

Comment:
this covers all file types scraped from s3:

# List all unique file extensions in the bucket
echo "Listing all unique file types in the bucket $BUCKET_NAME:"
aws s3 ls s3://$BUCKET_NAME/ --recursive | rev | awk -F '.' '{print $1}' | rev | grep -E '^[a-zA-Z0-9]{1,5}$' | sort | uniq

There is no magic number matching for json or html because they dont have magic numbers. We would need a less deterministic content based approach, like looking for opening and closing html tags and json brackets. Since this is an edge case we haven't actually even had an error for (we have only seen errors for tiffs without extensions), I think this can be lef tunresolved for now

Reply:
yeah this makes sense - in theory, we could assume if the first line is `<!DOCTYPE html>` it's HTML, or if the first char is `{` then it's JSON, but we can handle this if/when we get there since we've never seen these doc types in the networks

Note that all existing html/json in the buckets are probably files we've created